src/azul/indexer/document.py

```diff
@@ -361,6 +363,24 @@ class VersionType(Enum):
 C = TypeVar('C', bound=DocumentCoordinates)
 
 
+class SimpleSourceRef(SourceRef[SimpleSourceName, 'SimpleSourceRef']):
+
+    @classmethod
+    def from_aggregate_source(cls, source):
+        return cls(id=one(source['source'])['id'],
+                   name=SimpleSourceName(one(source['source'])['name']))
+
+    @classmethod
+    def from_contribution_source(cls, source):
+        return cls(id=source['source_id'], name=SimpleSourceName(source['source_name']))
+
+    # Stub is needed to aid PyCharm type hinting. Without this, instantiations
+    # of TDRSourceRef cause PyCharm to warn about the `name` parameter.
```

Comment:
TDRSourceRef is a copy-paste error here, but may also present where you copied this from.

Reply:
No longer relevant.